File: src/test/resources/verticle/failing.js
exports.vertxStartAsync = function(future) {
    var TestSuite = require('vertx-unit-js/test_suite');
    var suite = TestSuite.create("time_suite").test("timer_test", function(test) {
        var async = test.async();
        vertx.setTimer(50, function() {
            test.fail("the_failure");
        });
    });
    suite.run().resolve(future);
};

File: src/test/resources/verticle/failing.rb
require 'vertx-unit/test_suite'

def vertx_start_async start_future
  suite = VertxUnit::TestSuite.create('my_suite').test 'timer_test' do |context|
    async = context.async
    $vertx.set_timer 50 do
      context.fail Exception.new('the_failure')
    end
  end
  suite.run.resolve start_future
end